### DISCUSSION TITLE
Firefly in WSL (PHP slowness)

### DISCUSSION THREAD

**Discussion by justinshepard** [firefly-iii/firefly-iii#5051] [General]
I wasn't quite sure where to post this, but Discussions seems apt.

I had been banging my head against the desk trying to figure out why a self-hosted Firefly on WSL (Debian), php7.4 and nginx wasn't working. I could get as far as the register screen and then subsequent requests would just end up with a blank screen and a timeout error in the logs. Eventually I stumbled across [a StackOverflow answer](https://stackoverflow.com/a/50615652) that suggested adding `fastcgi_buffering off;` to the nginx config. I put this in my `server{}` section and suddenly it's lightning fast as expected.

Hopefully anyone else with this same issue will find this helpful. It might even be something James wants to mention in the installation docs/FAQs so that it's more visible.

Cheers!

**Comment by JC5** [comment]
Nice, thanks for sharing your solution. I've expanded the documentation by adding this to the FAQ :+1:
